spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/partitioner/ReplicaPartitioner.scala

```diff
@@ -0,0 +1,34 @@
+package com.datastax.spark.connector.rdd.partitioner
+
+import java.net.InetAddress
+
+import com.datastax.spark.connector.cql.CassandraConnector
+import org.apache.spark.{Partition, Partitioner}
+
+
+case class EndpointPartition(index: Int, endpoint: Set[InetAddress]) extends Partition
+
+class ReplicaPartitioner(partitionsPerReplicaSet: Int, connector:CassandraConnector) extends Partitioner {
+    val hosts = connector.hosts
```

Comment:
Discussion with Jacek, This needs to be simplified as the current code is far too complicated for the task at hand